app/models/user.rb

```diff
@@ -1,12 +1,16 @@
 class User < ApplicationRecord
-  has_many :posts, foreign_key: :author_id
-  has_many :comments, foreign_key: :author_id
-  has_many :likes, foreign_key: :author_id
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
+  has_many :posts, foreign_key: :author_id, dependent: :destroy
+  has_many :comments, foreign_key: :author_id, dependent: :destroy
+  has_many :likes, foreign_key: :author_id, dependent: :destroy
+
+  validates :name, presence: true, length: { maximum: 50 }
+  validates :post_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
 
```

Comment:
-  [ ] Here we should define the a function to check whether user is admin or not.
```ruby
def admin?
     role == 'admin' ? true : false
   end
```